### PR TITLE
Additional changes to ExpressionTransformer

### PR DESCRIPTION
**Pull request by yaeldekel**
This PR addresses the ExpressionTransformer part of issue #4610 and also addresses some left over comments from PR #4548 .

### CODE REVIEW

File: docs/api-reference/expression-estimator.md [hunk 1]
```diff
@@ -118,12 +118,12 @@ which uses the standard conversion parsing. When converting from a floating poin
...
-| ​long | ​convert to I8 | The input may be of any type. |
-| ​single, float | ​convert to R4 | ​The input may be of any type. |
-| ​double | ​convert to R8 | ​​The input may be of any type. |
-| ​text | ​convert to TX | ​​​The input may be of any type. This produces a default text representation. |
+| ​bool | ​convert to Boolean | The operand must be text or boolean. |
+| ​int | convert to <xref:System.Int32> | ​The input may be of any type. |
+| ​long | ​convert to <xref:System.Int64> | The input may be of any type. |
+| ​single, float | ​convert to <xref:System.Single> | ​The input may be of any type. |
+| ​double | ​convert to <xref:System.Double> | ​​The input may be of any type. |
+| ​text | ​convert to [text](xref:Microsoft.ML.Data.TextDataViewType) | ​​​The input may be of any type. This produces a default text representation. |
```
Comment (justinormont) [inline docs/api-reference/expression-estimator.md]: Addresses: https://github.com/dotnet/machinelearning/pull/4548#discussion_r361508517

#Resolved

File: src/Microsoft.ML.Transforms/Expression/BuiltinFunctions.cs [hunk 2]
```diff
@@ -969,11 +970,18 @@ public static I4 Pow(I4 a, I4 b)
...
+            if (a < 0)
+            {
+                a = -a;
+                neg = (b & 1) != 0;
+            }
 
             // Since the abs of the base is at least two, the exponent must be less than 31.
             if (b >= 31)
-                throw Contracts.Except("Cannot raise an integer to a power greater than 30");
+                return neg ? I4.MinValue : I4.MaxValue;
```
Comment (justinormont) [inline src/Microsoft.ML.Transforms/Expression/BuiltinFunctions.cs]: Addresses: https://github.com/dotnet/machinelearning/pull/4548#discussion_r360960548

#Resolved

File: src/Microsoft.ML.Transforms/Expression/TokKind.cs [hunk 3]
```diff
@@ -95,37 +95,6 @@ internal enum TokKind
...
-        Bittest,
-        Max,
-        Mean,
-        Response,
-        Norm,
-        FloatsFromBytes,
-        Param,
-        Auto,
-
-        // REVIEW: These are specific to the ExprTransform parser. Use a general mechanism.
```
Comment (justinormont) [inline src/Microsoft.ML.Transforms/Expression/TokKind.cs]: Addresses: https://github.com/dotnet/machinelearning/pull/4548#discussion_r360970284

#Resolved

File: src/Microsoft.ML.Transforms/ExpressionCatalog.cs [hunk 4]
```diff
@@ -17,6 +17,13 @@ public static class ExpressionCatalog
         /// This column's data type will be the same as that of the input column.</param>
         /// <param name="expression">The expression to apply to <paramref name="inputColumnNames"/> to create the column <paramref name="outputColumnName"/>.</param>
         /// <param name="inputColumnNames">The names of the input columns.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[Expression](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Expression.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
```
Comment (justinormont) [inline src/Microsoft.ML.Transforms/ExpressionCatalog.cs]: Addresses: https://github.com/dotnet/machinelearning/pull/4548#discussion_r361508772

#Resolved

**Comment by yaeldekel** [comment]
Thanks for the quick review!
Regarding vector column handling - this is mentioned in the .cs file (see line 47 in ExpressionTransformer.cs).
Regarding ONNX - I think it would be a fair amount of work, but definitely doable. I've opened issue #4615 to track this.

---
In reply to: [337484693](https://github.com/dotnet/machinelearning/pull/4614#issuecomment-337484693) [](ancestors = 337484693)

File: src/Microsoft.ML.Transforms/Expression/TokKind.cs [hunk 5]
```diff
@@ -95,37 +95,6 @@ internal enum TokKind
...
-        Pool,
-        Abs,
-        Bittest,
-        Max,
-        Mean,
-        Response,
-        Norm,
-        FloatsFromBytes,
-        Param,
-        Auto,
```
Comment (ganik) [inline src/Microsoft.ML.Transforms/Expression/TokKind.cs]: Why these are removed?

Comment (justinormont) [reply src/Microsoft.ML.Transforms/Expression/TokKind.cs]: See: https://github.com/dotnet/machinelearning/pull/4548#discussion_r360970284
These are used for the internal NN language, Net#. If we are no longer supporting that internally, then they are no longer needed. I don't think this PR provides the code for those functions, like convolution.

We may want to check that `abs(a)`, `min(a, b)`, `max(a,b )`, `sqrt(a)` still work.

**Pull request merged**